xchange-stream-ftx/src/main/java/info/bitrich/xchangestream/ftx/FtxStreamingService.java

```diff
@@ -2,55 +2,147 @@
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import info.bitrich.xchangestream.ftx.dto.FtxAuthenticationMessage;
 import info.bitrich.xchangestream.ftx.dto.FtxStreamRequest;
 import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
 import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
 import info.bitrich.xchangestream.service.netty.WebSocketClientCompressionAllowClientNoContextHandler;
 import io.netty.handler.codec.http.websocketx.extensions.WebSocketClientExtensionHandler;
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.knowm.xchange.ExchangeSpecification;
+import org.knowm.xchange.exceptions.ExchangeException;
+import org.knowm.xchange.ftx.service.FtxDigest;
+import org.knowm.xchange.utils.DigestUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.crypto.Mac;
+
 public class FtxStreamingService extends JsonNettyStreamingService {
 
   private static final Logger LOG = LoggerFactory.getLogger(FtxStreamingService.class);
   private final ObjectMapper mapper = StreamingObjectMapperHelper.getObjectMapper();
+  private boolean isLoggedIn = false;
+  private ExchangeSpecification exchangeSpecification = null;
 
   public FtxStreamingService(String apiUrl) {
     super(apiUrl);
   }
 
+  public FtxStreamingService(String apiUrl, ExchangeSpecification exchangeSpecification) {
+    super(apiUrl);
+    this.exchangeSpecification = exchangeSpecification;
+  }
+
```

Comment:
minor change but for consistency and to not have to store exchange spec in here, can you take a look at KrakenStreamingService and follow the pattern for creating the auth data, passing in a supplier?

Reply:
7de4dbb Just added a dto, used with supplier, avoid passing ExchangeSpecification into FtxStreamingService